### PR TITLE
Build sky_engine Dart SDK summary bundle on FlutterEngine.download().

### PR DESCRIPTION
**Pull request by scheglov**

### CODE REVIEW

File: packages/flutter_tools/lib/src/cache.dart [hunk 1]
```diff
@@ -234,13 +236,17 @@ class MaterialFonts {
...
 
   static const String kName = 'engine';
+  static const String kSkyEngine = 'sky_engine';
+  static const String kSkyServices = 'sky_services';
+  static const String kSdkBundle = 'sdk.ds';
 
   final Cache cache;
 
-  List<String> _getPackageDirs() => <String>['sky_engine', 'sky_services'];
+  List<String> _getPackageDirs() => const <String>[kSkyEngine, 'sky_services'];
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/cache.dart]: also use the `kSkyServices` constant here?

File: packages/flutter_tools/lib/src/cache.dart [hunk 2]
```diff
@@ -319,10 +328,23 @@ class FlutterEngine {
...
         await _downloadItem('Downloading package $pkgName...', url + pkgName + '.zip', pkgDir);
+        await pubGet(directory: pkgPath);
+      }
+    }
+
+    Status summaryStatus = logger.startProgress('Building SDK summary...');
+    try {
+      String skyEnginePath = path.join(pkgDir.path, kSkyEngine);
+      String skyServicesPath = path.join(pkgDir.path, kSkyServices);
+      if (!new File(path.join(skyEnginePath, kSdkBundle)).existsSync()) {
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/cache.dart]: nit: no braces for single line if statements

**Comment by devoncarew** [comment]
@pq, can you take a look?

File: packages/flutter_tools/lib/src/cache.dart [hunk 3]
```diff
@@ -319,12 +328,24 @@ class FlutterEngine {
...
-      Directory dir = new Directory(path.join(pkgDir.path, pkgName));
+      String pkgPath = path.join(pkgDir.path, pkgName);
+      Directory dir = new Directory(pkgPath);
       if (!dir.existsSync() || allDirty) {
         await _downloadItem('Downloading package $pkgName...', url + pkgName + '.zip', pkgDir);
+        await pubGet(directory: pkgPath);
       }
     }
 
+    Status summaryStatus = logger.startProgress('Building SDK summary...');
```
Comment (pq) [inline packages/flutter_tools/lib/src/cache.dart]: Maybe more specifically "Building Dart SDK summary..."?

Comment (scheglov) [inline packages/flutter_tools/lib/src/cache.dart]: Fixed.

**Comment by pq** [comment]
LGTM.

**Comment by eseidelGoogle** [comment]
FYI @johnmccutchan

**Comment by eseidelGoogle** [comment]
I'm curious how much time this adds to the flutter download step.  Doesn't matter hugely, but we currently display the total time (at least of our 17 pub gets), and I'm curious how much this increases it.

**Comment by scheglov** [comment]
```
scheglov@scheglov-macbookpro3:~/dart/flutter/bin/cache (master)$ flutter upgrade
Building flutter tool...
Downloading package sky_engine...                    0.4s
Running 'pub get' in sky_engine...                   0.3s
Downloading package sky_services...                  0.2s
Running 'pub get' in sky_services...                 0.9s
Building SDK summary...                              3.8s
Upgrading Flutter from /Users/scheglov/dart/flutter...
Already up-to-date.

Upgrading engine...
Already up-to-date.

Flutter • channel master • https://github.com/flutter/flutter.git
Framework • revision e880d51f00 (84 minutes ago) • 2016-08-16 08:56:31
Engine • revision fe509b0d96
Tools • Dart 1.19.0-dev.5.0
```

**Comment by eseidelGoogle** [comment]
Thanks!  Looks fantastic!

**Comment by devoncarew** [comment]
@scheglov, are you able to merge this?

**Comment by scheglov** [comment]
No, I'm not.
`Only those with write access to this repository can merge pull requests.`

**Comment by eseidelGoogle** [comment]
@scheglov I've sent you an invite for write access.  If any other dartians need it, please just let me know. :)

**Pull request merged**